linker-interface/shared/src/main/scala/org/scalajs/linker/interface/unstable/RuntimeClassNameMapperImpl.scala

```diff
@@ -15,6 +15,38 @@ package org.scalajs.linker.interface.unstable
 import org.scalajs.linker.interface.Semantics.RuntimeClassNameMapper
 
 object RuntimeClassNameMapperImpl {
-  def map(mapper: RuntimeClassNameMapper, className: String): String =
-    mapper(className)
+  case object KeepAll extends RuntimeClassNameMapper
+
+  case object DiscardAll extends RuntimeClassNameMapper
+
+  /* We use `pattern` and `flags` in the case parameters, rather than the
+    * `j.u.regex.Pattern`, because the latter does not have meaningful
+    * equality.
+    */
+  final case class RegexReplace(pattern: String, flags: Int,
+      replacement: String)(
+      val compiledPattern: java.util.regex.Pattern)
+      extends RuntimeClassNameMapper
+
+  final case class AndThen(first: RuntimeClassNameMapper,
+      second: RuntimeClassNameMapper)
+      extends RuntimeClassNameMapper
```

Comment:
In order to serialize different `RuntimeClassNameMapper`s, I needed to be able to know which kind of mapper I am serializing. However, this breaks the previous design that strongly hid the implementation details, and just passed a `RuntimeClassNameMapper` to the user as a black box that they could not peer into.

As discussed with @sjrd, this solution is not fantastic. I'm open to better solutions.

Reply:
An alternative would be to provide the serialization only (and no deserialization for that matter). Because all we seem to care about is to check that it is unchanged. So we can compare serialized forms. (I realize this would mean not relying on `sjsjsonnew`).

Reply:
Yes, that sounds like the right way to go. Moving these caused the tests to fail because of broken binary compatibility.

I could offer a serialization to a string of the whole linker config, store the serialized representation in a task instead of storing the object itself in the task, and then still rely on sbt's `.previous`. That does mean deleting the JSON format typeclasses for `StandardConfig`.

Reply:
It's for *serialization* that we need access to those things. *Deserialization* does not require those changes. So the alternative to only provide serialization does not make things any easier.

Reply:
Ah, that was maybe not clear from my comment. I meant to provide serialization in the linker interface directly. As such it would solve the problem of not exposing these things.

Reply:
Ah ! I hadn't thought about that. That's a funny way to look at the problem, but indeed, it seems to be a good approach.

Reply:
Thinking about this some more, it feels to me like this is the *only* correct approach for the entire `StandardConfig`. Otherwise adding a field to the `StandardConfig` becomes an incompatible change (because the serialization in sbt would not know about it and ignore it).